src/test/regress/sql/multi_extension.sql

```diff
@@ -55,62 +55,7 @@ DROP EXTENSION citus;
 
 SET citus.enable_version_checks TO 'false';
 
--- Create extension in oldest version
-CREATE EXTENSION citus VERSION '5.0';
-ALTER EXTENSION citus UPDATE TO '5.0-1';
-ALTER EXTENSION citus UPDATE TO '5.0-2';
-ALTER EXTENSION citus UPDATE TO '5.1-1';
-ALTER EXTENSION citus UPDATE TO '5.1-2';
-ALTER EXTENSION citus UPDATE TO '5.1-3';
-ALTER EXTENSION citus UPDATE TO '5.1-4';
-ALTER EXTENSION citus UPDATE TO '5.1-5';
-ALTER EXTENSION citus UPDATE TO '5.1-6';
-ALTER EXTENSION citus UPDATE TO '5.1-7';
-ALTER EXTENSION citus UPDATE TO '5.1-8';
-ALTER EXTENSION citus UPDATE TO '5.2-1';
-ALTER EXTENSION citus UPDATE TO '5.2-2';
-ALTER EXTENSION citus UPDATE TO '5.2-3';
-ALTER EXTENSION citus UPDATE TO '5.2-4';
-ALTER EXTENSION citus UPDATE TO '6.0-1';
-ALTER EXTENSION citus UPDATE TO '6.0-2';
-ALTER EXTENSION citus UPDATE TO '6.0-3';
-ALTER EXTENSION citus UPDATE TO '6.0-4';
-ALTER EXTENSION citus UPDATE TO '6.0-5';
-ALTER EXTENSION citus UPDATE TO '6.0-6';
-ALTER EXTENSION citus UPDATE TO '6.0-7';
-ALTER EXTENSION citus UPDATE TO '6.0-8';
-ALTER EXTENSION citus UPDATE TO '6.0-9';
-ALTER EXTENSION citus UPDATE TO '6.0-10';
-ALTER EXTENSION citus UPDATE TO '6.0-11';
-ALTER EXTENSION citus UPDATE TO '6.0-12';
-ALTER EXTENSION citus UPDATE TO '6.0-13';
-ALTER EXTENSION citus UPDATE TO '6.0-14';
-ALTER EXTENSION citus UPDATE TO '6.0-15';
-ALTER EXTENSION citus UPDATE TO '6.0-16';
-ALTER EXTENSION citus UPDATE TO '6.0-17';
-ALTER EXTENSION citus UPDATE TO '6.0-18';
-ALTER EXTENSION citus UPDATE TO '6.1-1';
-ALTER EXTENSION citus UPDATE TO '6.1-2';
-ALTER EXTENSION citus UPDATE TO '6.1-3';
-ALTER EXTENSION citus UPDATE TO '6.1-4';
-ALTER EXTENSION citus UPDATE TO '6.1-5';
-ALTER EXTENSION citus UPDATE TO '6.1-6';
-ALTER EXTENSION citus UPDATE TO '6.1-7';
-ALTER EXTENSION citus UPDATE TO '6.1-8';
-ALTER EXTENSION citus UPDATE TO '6.1-9';
-ALTER EXTENSION citus UPDATE TO '6.1-10';
-ALTER EXTENSION citus UPDATE TO '6.1-11';
-ALTER EXTENSION citus UPDATE TO '6.1-12';
-ALTER EXTENSION citus UPDATE TO '6.1-13';
-ALTER EXTENSION citus UPDATE TO '6.1-14';
-ALTER EXTENSION citus UPDATE TO '6.1-15';
-ALTER EXTENSION citus UPDATE TO '6.1-16';
-ALTER EXTENSION citus UPDATE TO '6.1-17';
-ALTER EXTENSION citus UPDATE TO '6.2-1';
-ALTER EXTENSION citus UPDATE TO '6.2-2';
-ALTER EXTENSION citus UPDATE TO '6.2-3';
-ALTER EXTENSION citus UPDATE TO '6.2-4';
-ALTER EXTENSION citus UPDATE TO '7.0-1';
```

Comment:
Why are we making these changes?

Reply:
> Why are we making these changes?

I had missed that we are also addressing #2834 here. What do you think moving these changes to a separate PR? My reason being that we can do better than concatenating files, and avoiding many of the `ALTER … ` commands there, which can also simplify some of the C code. For example some UDFs are called when some columns are added, so we needed to special handle those.

Reply:
Splitting up this PR makes sense. I can split this to another branch/PR, & then the optimizations you have in mind can be follow up commits